src/ui/popup.js

```diff
@@ -521,8 +535,8 @@ export default class Popup extends Evented {
             this._container = DOM.create('div', 'mapboxgl-popup', this._map.getContainer());
             this._tip       = DOM.create('div', 'mapboxgl-popup-tip', this._container);
             this._container.appendChild(this._content);
-            if (this.options.className) {
-                this.options.className.split(' ').forEach(name =>
+            if (this._classList) {
+                this._classList.forEach(name =>
```

Comment:
If we overwrote classes, then I think `this._container.className = [...this._classList.values()].join(' ')` would be the simplest way to accomplish this.